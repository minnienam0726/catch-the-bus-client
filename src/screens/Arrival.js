import React, { useEffect, useState } from "react";
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import { COLOR, FONTSIZE } from "../config/constants";

const Arrival = ({ route, navigation }) => {
  const [text, setText] = useState("");
  const [searchList, setSearchList] = useState([]);
  const { departure } = route.params;

  useEffect(() => {
    fetch(`${SERVER_URI}/search/station`, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ text }),
    })
      .then(async (response) => {
        const jsonResponse = await response.json();
        setSearchList(jsonResponse.payload);
      })
      .catch((error) => console.log(error));
  }, [text]);

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.screenContainer}>
        <TextInput
          style={styles.searchTextInput}
          placeholder="목적지를 검색하세요."
          onChangeText={setText}
        />
        <ScrollView style={styles.scrollView}>
          <View style={styles.researchResultContainer}>
            {searchList.map((station, key) => {
              return (
                <TouchableOpacity
                  style={styles.stationButton}
                  key={station}
                  onPress={() => {
                    navigation.navigate("Result", {
                      stations: {
                        departure: departure,
                        arrival: station,
                      },
                    });
                  }}
                >
                  <Text style={styles.stationText}>{station}</Text>
                </TouchableOpacity>
              );
            })}
          </View>
        </ScrollView>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  screenContainer: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    height: "100%",
  },
  searchTextInput: {
    width: "70%",
    borderWidth: 1,
    fontSize: 16,
  },
  scrollView: {
    width: "70%",
    backgroundColor: COLOR.GRAY,
    padding: 10,
  },
  researchResultContainer: {
    justifyContent: "space-around",
    width: "70%",
  },
  stationButton: {
    justifyContent: "center",
    alignItems: "center",
    height: 50,
    marginVertical: 5,
  },
  stationText: {
    color: COLOR.WHITE,
    fontSize: FONTSIZE.SMALL,
  },
});

export { Arrival };
