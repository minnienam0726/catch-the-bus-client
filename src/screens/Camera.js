import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import { GuideModal } from "../components/GuideModal";
import { CheckModal } from "../components/CheckModal";
import { SearchResultModal } from "../components/SearchResultModal";

import { COLOR } from "../api/constants";

const Camera = ({ navigation }) => {
  const [isGuideModalOpen, setIsGuideModalOpen] = useState(true);
  const [isCheckModalOpen, setIsCheckModalOpen] = useState(false);
  const [isSearchResultModalOpen, setIsSearchResultModalOpen] = useState(false);
  const [navigateHome, setNavigateHome] = useState(false);
  const [navigateArrival, setNavigateArrival] = useState(false);

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.screenContainer}>
        <Text style={styles.guideText}>더 가까이 촬영해주세요.</Text>
        <Text style={styles.cameraText}>카메라 화면이 삽입될 예정입니다.</Text>
        <TouchableOpacity
          style={styles.detectionButton}
          onPress={() => setIsCheckModalOpen(true)}
        >
          <Text style={styles.detectionButtonText}>인식하기</Text>
        </TouchableOpacity>
      </View>
      {navigateHome && navigation.goBack()}
      {navigateArrival && navigation.navigate("Arrival")}
      <GuideModal
        isGuideModalOpen={isGuideModalOpen}
        setIsGuideModalOpen={setIsGuideModalOpen}
      />
      <CheckModal
        isCheckModalOpen={isCheckModalOpen}
        setIsCheckModalOpen={setIsCheckModalOpen}
        setNavigateArrival={setNavigateArrival}
        setIsSearchResultModalOpen={setIsSearchResultModalOpen}
      />
      <SearchResultModal
        isSearchResultModalOpen={isSearchResultModalOpen}
        setIsSearchResultModalOpen={setIsSearchResultModalOpen}
        setNavigateHome={setNavigateHome}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  screenContainer: {
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    height: "100%",
    backgroundColor: COLOR.BLACK,
  },
  guideText: {
    width: "100%",
    marginVertical: 5,
    color: COLOR.RED,
    textAlign: "center",
    fontSize: 20,
  },
  cameraText: {
    color: COLOR.WHITE,
  },
  detectionButton: {
    justifyContent: "center",
    alignItems: "center",
    width: "50%",
    height: 70,
    marginVertical: 10,
    borderRadius: 100,
    backgroundColor: COLOR.BLUE,
  },
  detectionButtonText: {
    color: COLOR.WHITE,
    fontSize: 20,
  },
});

export { Camera };
