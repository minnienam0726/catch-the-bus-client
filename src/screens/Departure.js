import React, { useState } from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import Geolocation from "react-native-geolocation-service";

import { COLOR, FONTSIZE } from "../api/constants";

const Departure = ({ navigation }) => {
  const [geolocation, setGeolocation] = useState({});
  const [departure, setDeparture] = useState([]);

  const handleSubmit = () => {
    Geolocation.getCurrentPosition(
      (position) => {
        setGeolocation(position);
      },
      (error) => {
        console.log(error.code, error.message);
        setGeolocation(false);
      },
      { enableHighAccuracy: true, timeout: 15000, maximumAge: 10000 },
    );

    fetch(`${SERVER_URI}/geolocation`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ geolocation }),
    })
      .then(async (response) => {
        const jsonResponse = await response.json();
        setDeparture(jsonResponse.payload);
      })
      .catch((error) => console.log(error));
  };

  return (
    <SafeAreaView style={StyleSheet.container}>
      <View style={styles.screenContainer}>
        <View style={styles.researchResultContainer}>
          {departure.map((station, key) => {
            return (
              <TouchableOpacity
                style={styles.stationButton}
                key={station}
                onPress={() =>
                  navigation.navigate("Arrival", { departure: station })
                }
              >
                <Text style={styles.stationText}>{station}</Text>
              </TouchableOpacity>
            );
          })}
        </View>
        <TouchableOpacity
          style={styles.searchButton}
          onPress={() => handleSubmit()}
        >
          <Text style={styles.searchText}>내 주변 정류장 확인하기</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  screenContainer: {
    justifyContent: "space-around",
    alignItems: "center",
    width: "100%",
    height: "100%",
  },
  researchResultContainer: {
    justifyContent: "space-around",
    width: "70%",
  },
  stationButton: {
    justifyContent: "center",
    alignItems: "center",
    height: 60,
    marginVertical: 20,
    backgroundColor: COLOR.BLUE,
  },
  stationText: {
    color: COLOR.WHITE,
    fontSize: FONTSIZE.MEDIUM,
  },
  searchButton: {
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 100,
    width: "70%",
    height: 70,
    backgroundColor: COLOR.PINK,
  },
  searchText: {
    color: COLOR.BLACK,
    fontSize: FONTSIZE.MEDIUM,
  },
});

export { Departure };
