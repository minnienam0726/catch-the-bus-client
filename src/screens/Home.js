import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import Geolocation from "react-native-geolocation-service";

import { COLOR } from "../api/constants";

const Home = ({ navigation }) => {
  const [location, setLocation] = useState(false);
  const [geolocation, setGeolocation] = useState({});

  const getLocation = () => {
    Geolocation.getCurrentPosition(
      (position) => {
        setLocation(position);
      },
      (error) => {
        console.log(error.code, error.message);
        setLocation(false);
      },
      { enableHighAccuracy: true, timeout: 10000, maximumAge: 10000 },
    );
    setGeolocation(location);
  };

  const getStation = () => {
    fetch(SERVER_URL, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ geolocation }),
    })
      .then(async (response) => {
        const jsonResponse = await response.json();
        setGeolocation(jsonResponse.payload);
      })
      .catch((error) => console.log(error));
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.screenContainer}>
        <View style={styles.titleText}>
          <Text style={styles.titleSeoulText}>서울</Text>
          <Text style={styles.titleBusText}>버스 기사님,</Text>
          <Text style={styles.titleDestinationText}>*** 가시나요?</Text>
        </View>
        <TouchableOpacity
          style={styles.startButton}
          onPress={() => {
            navigation.navigate("TextDetection");
            getLocation();
            getStation();
          }}
        >
          <Text style={styles.startButtonText}>출발하기</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  screenContainer: {
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    height: "100%",
  },
  titleText: {
    width: "70%",
  },
  titleSeoulText: {
    paddingBottom: 5,
    textAlign: "left",
    fontSize: 20,
  },
  titleBusText: {
    textAlign: "left",
    fontSize: 40,
  },
  titleDestinationText: {
    paddingTop: 5,
    textAlign: "right",
    fontSize: 30,
  },
  startButton: {
    justifyContent: "center",
    alignItems: "center",
    width: "50%",
    height: 70,
    borderRadius: 100,
    backgroundColor: COLOR.BLUE,
  },
  startButtonText: {
    color: COLOR.WHITE,
    fontSize: 20,
  },
});

export { Home };
