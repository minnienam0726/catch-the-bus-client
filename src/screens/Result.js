import React, { useEffect, useState } from "react";
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import { COLOR, FONTSIZE } from "../api/constants";

const Result = ({ route, navigation }) => {
  const [boardingBus, setBoardingBus] = useState({});
  const [navigateHome, setNavigateHome] = useState(false);
  const [text, setText] = useState("");
  const [answer, setAnswer] = useState("");
  const { stations } = route.params;

  const busRoute = Object.entries(boardingBus);
  const busNumber = Object.keys(boardingBus);
  const searchNumber = busNumber.filter((number) => number == text);

  useEffect(() => {
    searchNumber[0] ? setAnswer("타세요!") : setAnswer("");
  }, [searchNumber]);

  useEffect(() => {
    fetch(`${SERVER_URI}/search/bus`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ stations }),
    })
      .then(async (response) => {
        const jsonResponse = await response.json();
        setBoardingBus(jsonResponse.payload);
      })
      .catch((error) => console.log(error));
  }, []);

  useEffect(() => {
    if (navigateHome) {
      return navigation.navigate("Home");
    }
  });

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.screenContainer}>
        <Text style={styles.message}>
          {busNumber ? "이 버스를 탑승하세요!" : "직행 버스가 없습니다!"}
        </Text>
        <ScrollView style={styles.scrollView}>
          <View>
            {busRoute.map((bus, key) => {
              return (
                <Text style={styles.resultText} key={bus[0]}>
                  버스 번호: {bus[0]} / 정류장 수: {bus[1]}
                </Text>
              );
            })}
          </View>
        </ScrollView>
        <TextInput
          style={styles.searchTextInput}
          placeholder="이 버스가 갈까요?"
          onChangeText={setText}
        />
        <Text style={styles.answerText}>{answer}</Text>
        <TouchableOpacity
          style={styles.homeButton}
          onPress={() => {
            setNavigateHome(true);
          }}
        >
          <Text style={styles.homeButtonText}>처음으로</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  screenContainer: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    height: "100%",
  },

  message: {
    margin: 20,
    color: COLOR.BLACK,
    fontSize: FONTSIZE.LARGE,
  },
  scrollView: {
    width: "70%",
    padding: 10,
    backgroundColor: COLOR.YELLOW,
  },
  resultText: {
    margin: 10,
    color: COLOR.BLACK,
    fontSize: FONTSIZE.SMALL,
  },
  searchTextInput: {
    width: "70%",
    margin: 10,
    borderWidth: 1,
    fontSize: 16,
  },
  answerText: {
    color: COLOR.RED,
    fontSize: FONTSIZE.SMALL,
  },
  homeButton: {
    justifyContent: "center",
    alignItems: "center",
    width: 150,
    height: 50,
    marginTop: 30,
    borderRadius: 100,
    backgroundColor: COLOR.BLUE,
  },
  homeButtonText: {
    color: COLOR.WHITE,
    fontSize: FONTSIZE.MEDIUM,
  },
});

export { Result };
