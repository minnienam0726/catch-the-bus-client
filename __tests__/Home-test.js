import React from "react";
import "react-native";
import { render } from "@testing-library/react-native";

import { Home } from "../src/screens/Home";

describe("Home 스크린", () => {
  it("Home 스크린 정상 렌더링", () => {
    const screen = render(<Home />);
    const json = screen.toJSON();

    expect(json).toMatchSnapshot();
  });

  it("제목 글씨", () => {
    const screen = render(<Home />);
    const titleSeoul = screen.getByText("서울");
    const titleBus = screen.getByText("버스 기사님,");
    const titleDestination = screen.getByText("*** 가시나요?");

    expect(titleSeoul).toBeTruthy();
    expect(titleBus).toBeTruthy();
    expect(titleDestination).toBeTruthy();
  });

  it("출발하기 버튼", () => {
    const screen = render(<Home />);

    expect(screen.getByTestId("startButton")).toBeTruthy();
  });
});
