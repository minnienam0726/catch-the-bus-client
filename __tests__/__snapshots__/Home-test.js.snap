// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 스크린 Home 스크린 정상 렌더링 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "paddingBottom": 5,
            "textAlign": "left",
          }
        }
      >
        서울
      </Text>
      <Text
        style={
          {
            "fontSize": 40,
            "textAlign": "left",
          }
        }
      >
        버스 기사님,
      </Text>
      <Text
        style={
          {
            "fontSize": 30,
            "paddingTop": 5,
            "textAlign": "right",
          }
        }
      >
        *** 가시나요?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#007AFF",
          "borderRadius": 100,
          "height": 70,
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="startButton"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 20,
          }
        }
      >
        출발하기
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
